packages/langium/test/grammar/type-system/inferred-types.test.ts

```diff
@@ -5,694 +5,331 @@
  ******************************************************************************/
 
 import { createLangiumGrammarServices, Grammar, EmptyFileSystem, expandToString, EOL } from '../../../src';
-import { collectAst, specifyAstNodeProperties } from '../../../src/grammar/type-system/ast-collector';
-import { collectAllAstResources, collectTypeResources } from '../../../src/grammar/type-system/type-collector/all-types';
-import { collectDeclaredTypes } from '../../../src/grammar/type-system/type-collector/declared-types';
-import { collectInferredTypes } from '../../../src/grammar/type-system/type-collector/inferred-types';
-import { AstTypes, InterfaceType, Property, PropertyType, UnionType } from '../../../src/grammar/type-system/type-collector/types';
-import { sortInterfacesTopologically } from '../../../src/grammar/type-system/types-util';
-import { parseHelper } from '../../../src/test';
-
-function describeTypes(name: string, grammar: string, description: (types: AstTypes) => void | Promise<void>): void {
-    describe(name, () => {
-        const types: AstTypes = {
-            interfaces: [],
-            unions: []
-        };
-        beforeAll(async () => {
-            const newTypes = await getTypes(grammar);
-            types.interfaces = newTypes.interfaces;
-            types.unions = newTypes.unions;
-        });
-        description.call(undefined, types);
-    });
-}
-
-describeTypes('inferred types of simple grammars', `
```

Comment:
I'm not sure about these testing rewrites here. Doing validation on types by checking formatted strings feels very brittle to me going forward. Can we still not get the types that are produced by the collector, before printing, and verify on those instead?

If we could have the existing test structure, verifying our internal representation, plus this extra checking on the generated output, that would also work well I think.

Reply:
To add as well, it's hard to follow if the integrity of the original tests is maintained with such a large refactor. I didn't get to look at it closely enough, but I would recommend a double check here by another colleague as well.

Reply:
The existing test structure was very tightly coupled with the exact layout of our old types. To prevent another test refactoring, I recommend that we only perform testing of types on the generated strings, as they contain all information that we get from the object model anyway.

Reply:
Okay. I'll take a closer look at the tests to verify things look good in the restructure.